Sources/Client/Client/Client.swift

```diff
@@ -147,20 +149,24 @@ public final class Client: Uploader {
     /// Creates a new instance of the network client.
     ///
     /// - Parameters:
-    ///   - config: The configuration object with details of how the new instance should be set up.
-    ///   - defaultURLSessionConfiguration: The base URLSession configuration `Client` uses for its
-    ///     URL sessions. `Client` is allowed to override the configuration with its own settings.
-    init(config: Client.Config, defaultURLSessionConfiguration: URLSessionConfiguration = .default) {
-        self.apiKey = config.apiKey
-        self.baseURL = config.baseURL
-        self.callbackQueue = config.callbackQueue ?? .global(qos: .userInitiated)
-        self.stayConnectedInBackground = config.stayConnectedInBackground
-        self.database = config.database
-        self.logOptions = config.logOptions
+    ///   - config: the configuration object with details of how the new instance should be set up.
+    ///   - defaultURLSessionConfiguration: the base URLSession configuration `Client` uses for its URL sessions.
+    ///                                     `Client` is allowed to override the configuration with its own settings.
+    ///   - defaultWebSocketProviderType: the default WebSocket provider type. `Client` will create it on set user.
+    init(config: Client.Config,
+         defaultURLSessionConfiguration: URLSessionConfiguration = .default,
+         defaultWebSocketProviderType: WebSocketProvider.Type = ProxyWebSocketProvider.self) {
```

Comment:
See https://getstream.slack.com/archives/CRB9P6L03/p1588842239493300

Reply:
For now, I'd like to initiate the type in the line as you did in your example and remove `ProxyWebSocketProvider`:
```swift
init(config: Client.Config,
    defaultURLSessionConfiguration: URLSessionConfiguration = .default,
    WebSocketProviderType: WebSocketProvider.Type = {
        ...
    }()
)
```